### PR TITLE
Migrate from github.com/ericchiang/k8s to github.com/kubernetes/client-go

### PR DESCRIPTION
**Pull request by sspaink**
The Kubernetes client project `ericchiang/k8s` has been archived since 2019, looks like `kubernetes/client-go` is the replacement. Migrated the plugin kube_inventory and Prometheus to use the new client.

### CODE REVIEW

**Review by Jagularr — APPROVED** [review]
Nice, makes some of the code cleaner too

File: plugins/inputs/kube_inventory/client.go [hunk 1]
```diff
@@ -4,115 +4,106 @@ import (
 	"context"
 	"time"
 
-	"github.com/ericchiang/k8s"
-	v1APPS "github.com/ericchiang/k8s/apis/apps/v1"
-	v1 "github.com/ericchiang/k8s/apis/core/v1"
-	v1beta1EXT "github.com/ericchiang/k8s/apis/extensions/v1beta1"
+	v1apps "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
```
Comment (srebhan) [inline plugins/inputs/kube_inventory/client.go:8]: Can you please decide for `<name>v1` **or** `v1<name>`. :-)

File: plugins/inputs/kube_inventory/persistentvolumeclaim_test.go [hunk 2]
```diff
@@ -6,8 +6,9 @@ import (
 	"testing"
 	"time"
 
-	"github.com/ericchiang/k8s/apis/core/v1"
-	metav1 "github.com/ericchiang/k8s/apis/meta/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (srebhan) [inline plugins/inputs/kube_inventory/persistentvolumeclaim_test.go:10]: Can't you just use `corev1`?

**Review by ssoroka — APPROVED** [review]

**Comment by varunjain0606** [comment]
@sspaink I am observing some issue related to the changes in this line and related logic. 
https://github.com/influxdata/telegraf/blob/1872037eb759d5ca82506592e53bd0d2c3df28d8/plugins/inputs/kube_inventory/node.go#L31 
I am getting a **string(val.Format)** value as either "**DecimalSI**" or "**BinarySI**" and hence getting error while parsing these values. 
Logs from parser:

2021-08-03T03:37:20Z D! [inputs.kube_inventory] failed to parse quantity: quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'
2021-08-03T03:37:20Z D! [inputs.kube_inventory] failed to parse quantity: quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'
2021-08-03T03:37:20Z D! [inputs.kube_inventory] failed to parse quantity: quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'
2021-08-03T03:37:20Z D! [inputs.kube_inventory] failed to parse quantity: quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'

Once I convert **string(val.Format)** to **val.String()**, things seem to work fine. May I know why format is used as input to
**convertQuantity()** function? Is it supposed to be like that? If yes then what could be the issue. 

Similar bug has been raised already. 
https://github.com/influxdata/telegraf/issues/9535

**Comment by sspaink** [comment]
@varunjain0606 Thank you for reporting this issue, looks like using format was a mistake. I noticed your pr #9581 to fix this, thank you for resolving it so quickly!

**Comment by varunjain0606** [comment]
@sspaink No problem. Glad to help. Cheers!

**Comment by gracewehner** [comment]
Hi @sspaink I found an issue related to the change in the watchPod() function in the file inputs/prometheus/kubernetes.go. I have created an issue #9600 and PR #9605 to address this.

**Pull request merged**